src/plugins/lightscript.js

```diff
@@ -121,7 +121,16 @@ pp.parseArrayComprehension = function (node) {
   const loop = this.startNode();
   node.loop = this.parseForStatement(loop);
   this.expect(tt.bracketR);
-  return this.finishNode(node, "ArrayComprehension");
+  node.comprehends = "array";
+  return this.finishNode(node, "Comprehension");
```

Comment:
On second thoughts I think I agree with you, worst case we can just do a type union in the ast visitor. I will use two different node types.